python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -1126,6 +1213,22 @@ def _add_assets_wipe_filter_to_query(
 
         return query
 
+    def get_asset_event_tags(self, asset_key: AssetKey) -> Sequence[Tuple[str, AbstractSet[str]]]:
+        tags = defaultdict(set)
+        query = (
+            db.select([AssetEventTagsTable.c.key, AssetEventTagsTable.c.value])
+            .distinct(AssetEventTagsTable.c.key, AssetEventTagsTable.c.value)
+            .where(AssetEventTagsTable.c.asset_key == asset_key.to_string())
+        )
```

Comment:
Is this based off of the run tags table query logic?  This was changed in https://github.com/dagster-io/dagster/pull/10036 because we were getting bad performance with it.

Reply:
Yep, it was. I updated this PR to follow the new changes in #10036!

Reply:
🎉